packages/ui/Button/hooks.ts

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react';
+import { ButtonColorTheme, ButtonIntent, ButtonShape } from './types';
+
+interface UseResolvedProps {
+  intent: ButtonIntent;
+  shape: ButtonShape;
+  theme?: ButtonColorTheme;
+  rounded?: 'md' | 'lg';
+}
```

Comment:
```suggestion
type UseResolvedProps = Pick<ButtonProps, pick_union>
```
새로 작성하지 않는게 좋을 것 같아요